src/js/view/controls/components/menu/category-button.ts

```diff
@@ -1,9 +1,14 @@
-import button from 'view/controls/components/button';
+import button, { Button } from 'view/controls/components/button';
 import { cloneIcon } from 'view/controls/icons';
 import { SimpleTooltip } from 'view/controls/components/simple-tooltip';
 import { createElement } from 'utils/dom';
+import { Tooltip, Menu } from 'types/generic.type';
 
-const categoryButton = (menu) => {
+type CategoryButton = Button & {
```

Comment:
ugh i keep waffling on interface vs type. here you _could_ do `interface CategoryButton extends Button { tooltip: Tooltip; }` but these are _stateless_ components right? so type makes more sense?

either way i'm fine with just thinking out loud

Reply:
I liked type because if I hover over Button in vscode as a type it shows all of the required properties and their types/call sigs, but as an interface I would have to click into the file. I prefer interface organizationally.